File: 007. Reverse Integer/reverse_integer.js
function reverseInteger(x) {
  const INT_MIN = -2147483648;
  const INT_MAX = 2147483647;

  let sign = x < 0 ? -1 : 1;
  x = Math.abs(x);

  let reversed = 0;
  while (x !== 0) {
    const digit = x % 10;
    x = Math.trunc(x / 10);
    reversed = reversed * 10 + digit;
  }

  reversed *= sign;

  if (reversed < INT_MIN || reversed > INT_MAX) {
    return 0;
  }
  return reversed;
}

function main() {
  console.log(reverseInteger(123));   // 321
  console.log(reverseInteger(-123));  // -321
  console.log(reverseInteger(120));   // 21
  console.log(reverseInteger(1534236469)); // 0 (overflow)
}

main();

File: 7. Reverse Integer/reverse_integer.py
def reverse_integer(x):
    INT_MIN = -2**31
    INT_MAX = 2**31 - 1

    sign = -1 if x < 0 else 1
    x = abs(x)

    reversed_num = 0
    while x != 0:
        digit = x % 10
        x //= 10
        reversed_num = reversed_num * 10 + digit

    reversed_num *= sign

    # Check overflow
    if reversed_num < INT_MIN or reversed_num > INT_MAX:
        return 0

    return reversed_num


# Pruebas locales
if __name__ == "__main__":
    print(reverse_integer(123))   # 321
    print(reverse_integer(-123))  # -321
    print(reverse_integer(120))   # 21
    print(reverse_integer(1534236469))  # 0 (overflow)
